lib/index.js

```diff
@@ -345,13 +345,20 @@ exports.deepEqual = function (obj, ref, seen) {
 
 // Remove duplicate items from array
 
-exports.unique = function (array, key) {
+exports.unique = function (array, keys) {
+
+    if (keys && !Array.isArray(keys)) {
+        keys = [keys];
+    }
 
     var index = {};
     var result = [];
 
     for (var i = 0, il = array.length; i < il; ++i) {
-        var id = (key ? array[i][key] : array[i]);
+        var id = (keys ? keys.reduce(function(id, key) {
```

Comment:
Reduce as others like it is not very fast, we should check it's not used in a critical path of hapi.